tests/Unit/DataStructures/Test_ComplexDataVector.cpp

```diff
@@ -0,0 +1,749 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <complex>
+#include <cstddef>
+#include <functional>
+#include <numeric>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/VectorImpl.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+#include "Utilities/StdHelpers.hpp"  // IWYU pragma: keep
+#include "tests/Unit/TestHelpers.hpp"
+
+// Heavily inspired by Test_DataVector.cpp
+// Nontrivial changes primarily in math tests: ComplexDataVectors are
+// mathematically different but not structurally different
+
+SPECTRE_TEST_CASE("Unit.DataStructures.ComplexDataVector",
```

Comment:
Is it really not possible to share some/most of these tests with DataVector? This also seems like a lot of copy-paste

Reply:
I agree - it is a ton of copy-paste. As a first start, I kept them silo-ed, but all aside from math operations could be directly duplicated, running once for each type. The math operations would need significantly more care (because reals and complex actually act differently mathematically), but I could work to unify them it if it's significantly preferable.